src/game-loop.js

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable no-console,default-case */
+
+import gameOfLife from './gol';
+import {
+  initializeRenderingGrid,
+  updateRenderingGrid,
+  renderColorfulBoard,
+  renderColorCheatSheet,
+} from './board-render';
+import { boards } from './boards';
+
+const readline = require('readline');
+
+const boardNames = Object.keys(boards);
+const renderZerosInBlack = process.argv.slice(2)[0] === '-b';
+const welcomeMessage = `@xxxx[{::::::::::::::::::::::::::::::::::>
+Game of Life
+  much more interesting than Game of Thrones
+@xxxx[{::::::::::::::::::::::::::::::::::>
+
+Press (a) to autorun, (t) to terminate
+`;
+
+let gameLoopInterval;
+let renderingSpeed;
+let selectedGameBoard;
+let boardRenderingGrid;
+let currentSystemState = 'init';
+let controlsLine = welcomeMessage;
+
+function displayControls() {
+  console.log(controlsLine);
+}
+
+function displayColorCheatSheet() {
+  console.log(`${renderColorCheatSheet()}\n`);
+}
+
+function displayBoard(board) {
+  if (boardRenderingGrid === undefined) {
+    boardRenderingGrid = initializeRenderingGrid(board);
+  }
+
+  boardRenderingGrid = updateRenderingGrid(boardRenderingGrid, board);
+
+  console.log(renderColorfulBoard(boardRenderingGrid, renderZerosInBlack));
+}
+
+function display(board) {
+  console.clear();
+
+
+  if (board !== undefined) {
+    displayBoard(board);
+    displayColorCheatSheet();
+  }
+
+  displayControls();
+}
+
+function generateBoardsQuestion() {
+  let counter = 0;
+  let question = 'Choose a board: ';
+
+  boardNames.forEach((boardName) => { question += `\n${counter}: ${boardName}`; counter += 1; });
```

Comment:
What about to use Array.reduce it looks like good option as you use accumulator (question) and iteration cycle (forEach).